### PR TITLE
Fix flaky `test_autolog_logs_signature_and_input_example`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Investigate this error:

https://github.com/mlflow/mlflow/runs/1653008314#step:4:493

## How is this patch tested?

- Fixed `test_autolog_logs_signature_and_input_example` + existing tests
- Ensure `test_autolog_logs_signature_and_input_example` is no longer flaky by running sklearn's cross version tests a few times.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/sklearn/test_sklearn_autolog.py [hunk 1]
```diff
@@ -863,8 +863,9 @@ def test_autolog_does_not_throw_when_mlflow_logging_fails(func_to_fail):
         mock_func.assert_called_once()
 
 
+@pytest.mark.parametrize("_", range(100))
```
Comment (harupy) [inline tests/sklearn/test_sklearn_autolog.py:866]: Run `test_autolog_logs_signature_and_input_example` 100 times to ensure the failure is flaky

**Comment by harupy** [comment]
I was able to repro the issue in this commit:

https://github.com/mlflow/mlflow/runs/1653916666#step:7:26

**Comment by harupy** [comment]
In the failed test,

```python
print(pyfunc_model.predict(input_example))
print(model.predict(X[:5]))
```

returns

```python
[[0.171504346208176  ]
 [0.34346150441640155]  <-
 [0.06895096846585114]  <-
 [0.05925789882165455]
 [0.03424907823290102]]
[[0.171504346208176  ]
 [0.3434615044164018 ]  <-
 [0.06895096846585136]  <-
 [0.05925789882165455]
 [0.03424907823290102]]
```

https://github.com/mlflow/mlflow/pull/3948/checks?check_run_id=1654268733#step:7:53

**Comment by harupy** [comment]
As a temporar workaround, we could use `assert_array_almost_equal` or `assert_allclose`.

File: tests/sklearn/test_sklearn_autolog.py [hunk 2]
```diff
@@ -881,7 +881,26 @@ def test_autolog_logs_signature_and_input_example(data_type):
...
+    #
+    # >>> model.predict(X[:5])
+    # [[0.171504346208176  ]
+    #  [0.3434615044164018 ]  <- diff
+    #  [0.06895096846585136]  <- diff
+    #  [0.05925789882165455]
+    #  [0.03424907823290102]]
+    #
+    # As a workaround, use `assert_array_almost_equal` instead of `assert_array_equal`
+    np.testing.assert_array_almost_equal(pyfunc_model.predict(input_example), model.predict(X[:5]))
```
Comment (harupy) [inline tests/sklearn/test_sklearn_autolog.py:903]: Replaced `assert_array_equal` with `assert_array_almost_equal`. Still not sure why this issue occurs.

**Review by dbczumar — APPROVED** [review]
Thanks @harupy !

**Pull request merged**